gino/ext/tornado.py

```diff
@@ -320,184 +224,17 @@ class Application(tornado.web.Application):
-
-        This function just takes the original generator
-        ``_execute.__wrapped__`` and manages to pass its futures to the
-        underlying asyncio loop.
-
-        It also calls ``_setup_connection`` and ``_teardown_connection``
-        methods and manages all errors that happen there.
-
-        """
-
-        self._transforms = transforms
-
-        try:
-            gen = super()._execute.__wrapped__(
-                self, transforms, *args, **kwargs
+        async def set_bind():
+            await self.set_bind(
+                dsn, echo=echo, min_size=pool_min_size, max_size=pool_max_size,
+                ssl=ssl, loop=asyncio_loop,
             )
 
-            data = None
-            exc_info = None
-
-            await self._setup_connection()
-
-            try:
-                while True:
-                    try:
-                        if exc_info is None:
-                            coro = gen.send(data)
-                        else:
-                            coro = gen.throw(*exc_info)
-                    except StopIteration:
-                        return
-                    exc_info = None
-                    data = None
-                    # noinspection PyBroadException
-                    try:
-                        data = await coro
-                    except:  # NOQA
-                        exc_info = sys.exc_info()
-            finally:
-                await self._teardown_connection()
-        except Exception as e:
-            # noinspection PyBroadException
-            try:
-                self._handle_request_exception(e)
-            except Exception:
-                tornado.log.app_log.exception("exception in exception handler")
-        finally:
-            if self._prepared_future is not None:
-                if not self._prepared_future.done():
-                    self._prepared_future.set_result(None)
-
-
-# noinspection PyAbstractClass
-class GinoRequestHandler(AsyncioRequestHandler):
-    """
-    Base class for all request handlers that use GINO.
-
-    In addition to features provided by :py:class:`~.AsyncioRequestHandler`,
-    this class manages lazy connections for each request.
-
-    """
-
-    __db_connection = None
-
-    @property
-    def use_connection_for_request(self):
-        """
-        If ``True``, a lazy connection is created for each request.
-
-        That is, whenever the first query occurs, a new connection is
-        borrowed from the application's db object. All succeeding queries made
-        within this request will reuse that connection. The connection will be
-        returned to the pool once the request is finished or the
-        :py:meth:`~.release_connection()` method is called explicitly.
-
-        This property is equal to
-        :py:attr:`Application.use_connection_for_request`
-        by default.
-
-        """
-        return self.application.use_connection_for_request
-
-    @property
-    def db(self):
```

Comment:
Also, this was a very useful property (saves a bit of typing 😉).

Reply:
Right, we can have a RequestHandler subclass with some convenient methods. Users need to inherit tornado's anyway, and they can choose not to use.